cmd/src/actions_exec.go

```diff
@@ -401,14 +401,49 @@ func prepareAction(ctx context.Context, action Action) error {
 // have been pulled from or pushed to a registry. See
 // https://windsock.io/explaining-docker-image-ids/ under "A Final Twist" for a good
 // explanation.
-func getDockerImageContentDigest(ctx context.Context, image string) (string, error) {
+func getDockerImageContentDigest(ctx context.Context, image string, logger *actionLogger) (string, error) {
 	// TODO!(sqs): is image id the right thing to use here? it is NOT the
 	// digest. but the digest is not calculated for all images (unless they are
 	// pulled/pushed from/to a registry), see
 	// https://github.com/moby/moby/issues/32016.
 	out, err := exec.CommandContext(ctx, "docker", "image", "inspect", "--format", "{{.Id}}", "--", image).CombinedOutput()
 	if err != nil {
-		return "", fmt.Errorf("error inspecting docker image (try `docker pull %q` to fix this): %s", image, bytes.TrimSpace(out))
+		// An error could mean the image just didn't exist, so we need to pull it first.
+		logger.Infof("Pulling image %q...\n", image)
+		pullCmd := exec.CommandContext(ctx, "docker", "image", "pull", image)
+		stdout, err := pullCmd.StdoutPipe()
+		if err != nil {
+			return "", fmt.Errorf("error pulling docker image %q: %s", image, err)
+		}
+		stderr, err := pullCmd.StderrPipe()
+		if err != nil {
+			return "", fmt.Errorf("error pulling docker image %q: %s", image, err)	
+		}
+		outScanner := bufio.NewScanner(stdout)
+		go func() {
+			for outScanner.Scan() {
+				logger.Infof("%s\n", outScanner.Text())
+			}
+		}()
+		errScanner := bufio.NewScanner(stderr)
+		go func() {
```

Comment:
Instead of doing this whole dance with `StdoutPipe`/`StderrPipe` and the scanners, I think we can simplify this by moving some of that into the logger:

```go
stdout := logger.InfoPipe()
pullCmd.Stdout = stdout
```

The `InfoPipe` and `WarnPipe` methods could return a prefixed `io.Writer` that **only logs if the verbose mode is turned on**.

Take a look at this method to get some ideas:

https://github.com/sourcegraph/src-cli/blob/1266471f6ceb8c23d527bc068799ba7913971aee/cmd/src/actions_exec_logger.go#L144-L160